### ISSUE TITLE
Consider using BLIS

### ISSUE THREAD

**Issue by bluss** [bluss/rust-ndarray#171]
https://github.com/flame/blis

It has the potential to fit much better since they support flexible memory layouts (just like ndarray).

**Comment by bluss** [comment]
BLIS is a super interesting project. They and their documentation and papers inspired [matrixmultiply](https://github.com/bluss/matrixmultiply) which put pure rust support for matrix multiplication in a much better place.

As it is now BLIS is not something I want to integrate with in ndarray. It is yet far from the mature status as other linear algebra projects you can pick off the shelf (old BLAS stuff).

BLIS has

- high overhead for small problems
- requires manual config for building for a microarch
- requires manual config for choosing number of threads

in short, it's a project with different aims than ndarray right now.
